rust/gitxetcore/src/git_integration/migration/data_import.rs

```diff
@@ -0,0 +1,158 @@
+        let ret_data
+        = pft.clean_file_and_report_progress(
+            &PathBuf::from_str(&name).unwrap(),
+            git_lfs_reader,
+            &Some(progress_reporting),
+        )
+        .await.map_err(|e| {
+            warn!("Error filtering git-lfs blob {name}: {e}, contents = \"{}\".  Importing as is.",
+            std::str::from_utf8(&src_data[..]).unwrap_or("<Binary Data>"));
+            e
+        }).unwrap_or(src_data);
+
+        Ok(ret_data)
+    } else if is_xet_pointer_file(&src_data[..]) {
+        info!("Source blob ID {blob_oid:?} is git xet pointer file; smudging through git-xet.");
+        let git_xet_pointer = smudge_git_xet_pointer(src_repo_dir, src_data.clone()).await?;
+        let ret_data = pft
+            .clean_file_and_report_progress(
+                &PathBuf::from_str(&name).unwrap(),
+                git_xet_pointer,
+                &Some(progress_reporting),
+            )
+            .await
+            .map_err(|e| {
+                warn!(
+                    "Error filtering Xet pointer at {name}, contents = \"{}\".  Importing as is.",
+                    std::str::from_utf8(&src_data[..]).unwrap_or("<Binary Data>")
+                );
+                e
+            })
+            .unwrap_or(src_data);
+
+        Ok(ret_data)
+    } else {
+        debug!("Cleaning blob {blob_oid:?} of size {}", src_data.len());
+        // Return the filtered data
+        pft.clean_file_and_report_progress(
+            &PathBuf::from_str(&name).unwrap(),
+            src_data,
+            &Some(progress_reporting),
+        )
+        .await
+    }
+}
+
+async fn ensure_git_lfs_is_initialized(
+    src_repo_dir: &Path,
+    init_locks: &Arc<RepoInitTracking>,
+) -> Result<()> {
+    if !init_locks
+        .lfs_is_initialized
+        .load(std::sync::atomic::Ordering::Relaxed)
+    {
+        let _lg = init_locks.git_lfs_init_lock.lock().await;
+
+        if init_locks
+            .lfs_is_initialized
+            .load(std::sync::atomic::Ordering::Relaxed)
+        {
+            return Ok(());
+        }
+
+        info!("Running git lfs install in {src_repo_dir:?}");
+
+        run_git_captured(Some(&src_repo_dir.to_path_buf()), "lfs", &["install"], true, None).map_err(|e| {
+            error!("Error running `git lfs install` on repository with lfs pointers: {e:?}.  Please ensure git lfs is installed correctly.");
+            e
+        })?;
+
+        init_locks
+            .lfs_is_initialized
+            .store(true, std::sync::atomic::Ordering::SeqCst);
+    }
+
+    Ok(())
+}
+
+pub fn is_git_lfs_pointer(data: &[u8]) -> bool {
+    if data.len() >= 1024 {
+        return false;
+    }
+
+    // Convert &[u8] to a &str to parse as text.
+    let Ok(text) = std::str::from_utf8(data) else {
+        return false;
+    };
+
+    let text = text.trim_start();
+
+    text.starts_with("version https://git-lfs.github.com/spec/v1") && text.contains("oid sha256:")
```

Comment:
Are there other headers for lfs pointer files? e.g. for an older spec, or without the github.com title?

Reply:
Nope, this is it.